test/cql-pytest/test_type_date.py

```diff
@@ -56,17 +56,20 @@ def test_type_date_from_int_prepared(cql, table1):
 # one that doesn't fit 32 bits, is an error. Cassandra says:
 # "Unable to make unsigned int (for date) from: '-123'".
 # Reproduces #17066.
-@pytest.mark.xfail(reason="#17066")
 def test_type_date_from_int_unprepared_underflow(cql, table1):
     p = unique_key_int()
-    with pytest.raises(InvalidRequest, match="unsigned int"):
+    with pytest.raises(InvalidRequest, match="overflows"):
```

Comment:
As the comment explains, Cassandra's error message in this case is "Unable to make unsigned int (for date) from: '-123'", so it won't match you changed string "overflows". Please make sure you run your test also on Cassandra (`test/cql-pytest/run-cassandra test_type_date.py::test_type_date_from_int_unprepared_underflow`) to make sure that the test itself is correct and passes on Cassandra.

Also I don't think the word "overflows" is a good choice, because the error can be that the number is either too low (negative) or two high, and the word "overflow" makes it sound like it's only the latter. You can copy Cassandra's error message (see above) if you don't want to be too original in the error message :-)